pkg/util/containers/filter_test.go

```diff
@@ -483,11 +485,12 @@ func TestNewMetricFilterFromConfig(t *testing.T) {
 }
 
 func TestNewAutodiscoveryFilter(t *testing.T) {
-	resetConfig()
+	cfg := mock.New(t)
+	resetConfig(cfg)
```

Comment:
As the `cfg` mock config is freshly created. It should already be “clean” and `resetConfig` shouldn’t be needed.

Reply:
This instance can be removed since the config is new here, but it's necessary for the later calls.